### PR TITLE
Makes the webinos platform npm installable

### PR DESCRIPTION
**Pull request by eelcocramer**
Jira issue: WP-100

### CODE REVIEW

**Comment by andreasbotsikas** [comment]
Hi @eelcocramer, I am not sure if we should add the webinos_build too. Since we are making it npm installable, we shouldn't be moving staff on our own but rather let npm place it in the path (next to node.js)... @habibvirji  do you think we can totally remove the webinos_build script? The only added value of this script is that it runs the closure compiler and generates the webinos.js file. Am I right?

**Comment by eelcocramer** [comment]
It makes sense not to include the webinos build in /usr/local/bin but instead we can add an install step in the npm install that runs the closure compiler and generates the js file.

**Comment by habibvirji** [comment]
I think so that will be perfect. @andreas yes it just creates webinos.js and installs webinos pzp globally. 

Cheers
Habib Virji

**Comment by eelcocramer** [comment]
Ok @andreasbotsikas and @habibvirji I've just pushed the changes to make it so. Note that webinos_build.js is now changed to only generate the webinos.js file. It is called automatically when calling 'npm install [-g]'. 

If this change is pulled in we should also update the pages on the developer portal.

@andreasbotsikas can you test this change on Windows?

**Comment by andreasbotsikas** [comment]
Ok so my only hesitation before merging is that now we do need java in path in order to do an install. This is not bad as in order to generate the final webinos.js file we do need java to run the closure compiler and if you don't have it, pzp is useless. Could @eelcocramer add a jira issue to make the concatenation from node so that we get rid of this dependency later on?

**Comment by eelcocramer** [comment]
I've created a jira issue to get rid of the java dependency.

http://jira.webinos.org/browse/WP-461

File: package.json [hunk 1]
```diff
@@ -24,8 +24,10 @@
   , "ansi" : "0.0.x"
   , "mdns" : "0.x.x"
   , "now" : "0.8.1"
-  , "node-uuid" : "x.x.x"
+  , "node-uuid" : "1.3.x"
```
Comment (andreasbotsikas) [inline package.json:27]: Why do we need this twice?

File: package.json [hunk 2]
```diff
@@ -24,8 +24,10 @@
...
   , "find-dependencies" : "~0.0.2"
+  , "node-uuid" : "1.3.x"
+  , "find-dependencies" : "0.0.x"
```
Comment (andreasbotsikas) [inline package.json:30]: I think these were merged by mistake... We already have the proper node_modules declared...

File: package.json [hunk 3]
```diff
@@ -35,4 +37,11 @@
   , "npm" : ">= 1.1.17"
   }
 , "preferGlobal": true
+, "bin":
+  { "webinos_pzh": "webinos_pzh.js"
+  , "webinos_pzp": "webinos_pzp.js"
+  }
+, "scripts":
+  { "install": "node webinos_build.js"
```
Comment (andreasbotsikas) [inline package.json:45]: I hope this doesn't disrupt the normal npm install which also build the native modules, right?

**Pull request merged**